credentials/tls.go

```diff
@@ -33,6 +33,7 @@ import (
 // It implements the AuthInfo interface.
 type TLSInfo struct {
 	State tls.ConnectionState
+	CommonAuthInfo
```

Comment:
Good point. `CheckSecurityLevel` is updated as suggested. Regarding the comment - "The zero-value...", I provided a constructor for `CommonAuthInfo` that takes security level as an argument and does some sanity check inside to make sure `NoSecurity` will not be used.

Reply:
Sorry, I am a little confused here. In your first comment you suggested "the best solution is to make the zero value illegal..." and in the comment above you suggested `NoSecurity should be valid in a CommonAuthInfo`..

Reply:
Ah I see. your first comment is applied in a context where `TransportCredentials` is used in which case we should not have a `NoSecurity` level. But if we have local credential in future (not sure if it will implement `TransportCredentials`), the security level will be `PrivacyAndIntegrity` for UDS and `NoSecurity` for TCP loopback.